gnocchi/metric/v1/measures/results.go

```diff
@@ -0,0 +1,84 @@
+package measures
+
+import (
+	"encoding/json"
+
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// Measure is an datapoint thats is composed with a timestamp and a value.
+type Measure struct {
+	// TimeStamp represents a timestamp of when measure was pushed into the Gnocchi.
+	TimeStamp string `json:"-"`
+
+	// Granularity is a level of precision that is kept when aggregating data.
+	Granularity float64 `json:"-"`
+
+	// Value represents a value of data that was pushed into the Gnocchi.
+	Value float64 `json:"-"`
+}
+
+/*
+UnmarshalJSON helps to unmarshal response from reading Gnocchi measures.
+
+Gnocchi APIv1 returns measures in a such format:
+
+[
+    [
+        "2017-01-08T10:00:00+00:00",
+        300.0,
+        146.0
+    ],
+    [
+        "2017-01-08T10:05:00+00:00",
+        300.0,
+        58.0
+    ]
+]
+
+Helper unmarshals every nested array into the Measure type.
+*/
+func (r *Measure) UnmarshalJSON(b []byte) error {
+	type tmp Measure
+	var measuresSlice []interface{}
+
+	var s struct {
+		tmp
+	}
+	err := json.Unmarshal(b, &measuresSlice)
+	if err != nil {
+		return err
+	}
+
+	*r = Measure(s.tmp)
+	r.TimeStamp = measuresSlice[0].(string)
```

Comment:
Additonally, each of the needs to check the assumed data type:
```go
var ok bool
r.TimeStamp, ok = measuresSlice[0].(string); !ok {
  // return an error because it's not a string
}
```

Without those checks, the code can `panic`. Remember, OpenStack is returning data that could be anything. Just because it's `[string, float64, float64]` now doesn't mean it won't change in the future, and if it does, we don't want programs to crash because of it.